src/date-input/date-input.tsx

```diff
@@ -24,21 +27,32 @@ export const DateInput = ({
     onBlurRaw,
     readOnly,
     id,
+    type,
+    rangeValue,
+    setIsOpenCalendar,
+    setFocusTo,
+    setTransitionValue,
+    transitionValues,
+    hoverValue,
+    focusTo,
+    handleCancelButton,
+    calendarRootId,
     ...otherProps
-}: DateInputProps) => {
+}: DateInputProps & DateInputRangeProps) => {
     // =============================================================================
     // CONST, STATE, REF
     // =============================================================================
     const [dayValue, _setDayValue] = useState<string>("");
     const [monthValue, _setMonthValue] = useState<string>("");
     const [yearValue, _setYearValue] = useState<string>("");
     const [currentFocus, _setCurrentFocus] = useState<FieldType>("none");
+    const [calendarManualInput, setCalendarManualInput] = useState<string>("");
```

Comment:
I have been looking through and find that using a string requires splitting and checking, which is laborious. Also curious as to the issue of assigning to the existing `dayValue` etc states?

Is it that any selection of within the calendar is not confirmed until you press ok? If so, keeping it as an object/separate variables might be easier